helpers.go

```diff
@@ -870,6 +884,55 @@ func (ctx *e2econtext) summarizeSuiteFindings(t *testing.T, suite string) {
 		}
 		t.Logf("Scan %s contained %d checks that failed, but have a remediation available", scan.Name, len(failedWithRemediationList.Items))
 	}
+	// print out the profile assertion file if it doesn't exist
+	if !ctx.profileAssertExist && ctx.profileAssert.RuleResults != nil {
+		profileTestBytes, err := yaml.Marshal(ctx.profileAssert)
+		if err != nil {
+			t.Fatalf("failed to marshal profile assertion file: %s", err)
+		}
+		t.Logf("Profile assertion file:\n%s", string(profileTestBytes))
+	}
+}
+
+func (ctx *e2econtext) assertProfileAssertionFile(t *testing.T, resultList *cmpv1alpha1.ComplianceCheckResultList, afterRemediations bool) {
+	profileTestFilePath := path.Join(ctx.rootdir, "tests", "assertions", ctx.platform, ctx.product+"-"+ctx.Profile+"-"+ctx.version+".yml")
+	_, err := os.Stat(profileTestFilePath)
+	if os.IsNotExist(err) {
+		t.Logf("failed to find profile assertion file %s", profileTestFilePath)
+	}
+
+	buf, err := os.ReadFile(profileTestFilePath)
+
+	// do not fail on not found, we will print out the file according to the test result
+	if err != nil && !os.IsNotExist(err) {
```

Comment:
nit: We should be safe from the second half of this conditional since we've already checked that on line 900.

Reply:
Oh - I suppose the only thing we're doing on line 900 is logging, and not handling test execution.